### ISSUE TITLE
Interoperability with state restoration

### ISSUE THREAD

**Issue by foodchaining** [rrousselGit/flutter_hooks#164]
Hi,
There is a [state restoration](https://flutter.dev/go/state-restoration-design) facility upcoming to flutter. Will `flutter_hooks` be compatible with it?

**Comment by rrousselGit** [comment]
For now, you can use `StatefulHookWidget` and the official API for restorable state

A hook support may need some PR

**Comment by blaugold** [comment]
Hey @rrousselGit and @foodchaining,

I've created a PR (#167) to add support for managing `RestorableProperty` with a hook. Would love to get some feedback whether that's a good direction.

**Comment by rrousselGit** [comment]
@knaeckeKami In your example, is there any way to avoid that addPostFrameCallback in the `restoreState`?

One core problem behind this feature is, we want the initial `build` to already be restored.

**Comment by knaeckeKami** [comment]
Are you talking about the addPostFrameCallback in the example?

This is only needed to show the Snackbar in case the state was restored, the first build would already have the correct restored state.

**Comment by knaeckeKami** [comment]
Yes, it's definitely verbose, it's basically a full port of the RestorationMixin API. In the `useRestoration` callback you can do things like unregister previously registered properties or change the restoration id using the didUpdateRestorationId method. 

Most use cases probably don't need that, so it might make sense to trim the API down. 

It's just a first draft of what state restoration using hooks might look like, hence the experimental status, I'm happy for feedback on what could be improved in the API.

**Comment by knaeckeKami** [comment]
Thanks! I'll work on refining this later, might take a couple of weeks though.
